src/main/kotlin/exercise2/task3/ConvertMethodToFunctionalApproach.kt

```diff
@@ -17,24 +17,25 @@ import kotlin.math.abs
  */
 
 internal fun List<Int>.findPairWithBiggestDifference(): Pair<Int, Int> {
-    // TODO refactor me to functional approach and make tests pass!!!
-    var resultPair: Pair<Int, Int>? = null
-    var biggestDifference = Int.MIN_VALUE
-
-    for (i in this.indices) {
-        for (j in (i + 1) until this.size) {
-            val first = this[i]
-            val second = this[j]
-            val absDifference = abs(first - second)
-
-            if (absDifference >= biggestDifference) {
-                biggestDifference = absDifference
-                resultPair = Pair(first, second)
-            }
-        }
-    }
 
-    return resultPair!!
+//    var resultPair: Pair<Int, Int>? = null
+//    var biggestDifference = Int.MIN_VALUE
+//
+//    for (i in this.indices) {
+//        for (j in (i + 1) until this.size) {
+//            val first = this[i]
+//            val second = this[j]
+//            val absDifference = abs(first - second)
+//
+//            if (absDifference >= biggestDifference) {
+//                biggestDifference = absDifference
+//                resultPair = Pair(first, second)
+//            }
+//        }
+//    }
+//    return resultPair!!
+
+    return this.sorted().let { it.first() to it[lastIndex] }
```

Comment:
You made a mistake in the implementation by stating the `lastIndex`. In the code you wrote, the `lastIndex` is the last index of the receiver object list. It's the same as if we wrote `this.lastIndex`.

TLDR: you should use `it.lastIndex` instead of `this.lastIndex`

In this case, the result will be correct, but if we, for example, filter elements of the list after sorting it, the `lastIndex` of the list we are operating on would change, and you would get an `IndexOutOfBounds` error. 

Let's show this visually. In the following example, we have the list `[3, 2, 4, 5, 1]`.
After calling `sorted()` on the list, we would get a new list `[1, 2, 3, 4, 5]`, with the last index value being `4`.
If after that, on the result list we call `filter { it > 2 }`, the new result list would be `[3, 4, 5]`, where now the value of the last index is `2`, and if we would call `it[this.lastIndex]` as in your code, we would get `IndexOutOfBoundsException` because `this.lastIndex` value is 4, which is greater than current last index.